New geometry class and some minor fixes

bwohlberg opened the pull request:
- Add a `RegularPolygon` geometric entity.
- Fix some typos and formatting issues in the API docs.
- Add some missing attributes in `setup` in `setup.py`.

Note: the API docs for `RegularPolygon.__init__` are specified in a way that is not consistent with the rest of the package because I could not see a consistent way to include them.

pep8speaks commented:
Hello @bwohlberg! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`src/xdesign/geometry/area.py`](https://github.com/tomography/xdesign/blob/78fdf393aac74e9857f43879152eb24ab8d69533/src/xdesign/geometry/area.py):

> [Line 557:80](https://github.com/tomography/xdesign/blob/78fdf393aac74e9857f43879152eb24ab8d69533/src/xdesign/geometry/area.py#L557): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)

carterbox commented:
Thanks for opening a PR, I have seen it, and I will take a look when I have time.

bwohlberg commented:
I have not done any testing to confirm, but I expect that 31eb3cf will resolve #77.

carterbox reviewed (changes requested):
Please add some text to the docs for RegularPolygon about the default orientation. i.e. Something about how one vertex is always located at `(center + radius, center)` unless that angle parameter is used.

Otherwise, LGTM! Thanks for cleaning up the docs!

bwohlberg commented:
> Please add some text to the docs for RegularPolygon about the default orientation. i.e. Something about how one vertex is always located at `(center + radius, center)` unless that angle parameter is used.

Added. I was unable to check whether the text is formatted properly because the sphinx configuration doesn't actually generate the docs for `__init__`.

I also added a setting for `bibtex_bibfiles` sphinx `conf.py` in an attempt to address the failure of the readthedocs build workflow.

carterbox commented on src/xdesign/geometry/area.py:
```diff
@@ -533,6 +540,42 @@ def contains(self, other):
+    polygon is oriented so that one of the vertices is at coordinates
+    :math:`(x + r, x)` where :math:`x` is the x-coordinate of
+    ``center`` and :math:`r` = ``radius``. The ``angle`` parameter is
+    only meaningful modulo :math:`\pi /` ``order`` since rotation by
+    :math:`\pi /` ``order`` gives a result equivalent to no rotation.
```
```suggestion
    only meaningful modulo :math:`2\pi /` ``order`` since rotation by
    :math:`2\pi /` ``order`` gives a result equivalent to no rotation.
```
Modulo `2PI / order`? Because there are 2PI radians in one rotation.

bwohlberg replied:
Yes, good catch. Fix pushed.

carterbox commented:
> I also added a setting for bibtex_bibfiles sphinx conf.py in an attempt to address the failure of the readthedocs build workflow.

Yes, I'm trying to fix this problem in #81. I tried to setup readthedocs to validate pull requests, but the docs it generates fail with some SSL certificate error for me.
https://xdesign--78.org.readthedocs.build/en/78/

bwohlberg commented:
> Yes, I'm trying to fix this problem in #81. I tried to setup readthedocs to validate pull requests, but the docs it generates fail with some SSL certificate error for me. https://xdesign--78.org.readthedocs.build/en/78/

I can access that URL without any issues. (Assuming I'm understanding the problem correctly.)

carterbox commented:
OK, then it's being blocked by Argonne's cybersecurity policy, so I can't see it from my work station. Please look to see if the docstring is formatted how you like it.

bwohlberg commented:
It looks good to me.

carterbox commented:
Thanks for your contribution! 🔥
